src/renderer/components/Price.js

```diff
@@ -57,8 +58,12 @@ export default function Price({
     value: valueNum,
     disableRounding: true,
   });
-  const counterValue =
-    typeof rawCounterValue !== "undefined" ? BigNumber(rawCounterValue) : rawCounterValue;
+
+  const counterValue = rate
+    ? rate.times(valueNum) // NB Allow to override the rate for swap
```

Comment:
Should it do .integerValue() at the end? Because I'm not sure that the formatter will 🤔

Reply:
What do you mean? I only added the `rate` part but further down the code still uses `counterValue` as a BigNumber and is expected as such in `CurrencyUnitValue`